testng-eclipse-plugin/src/main/org/testng/eclipse/ui/util/ConfigurationHelper.java

```diff
@@ -373,6 +379,42 @@ else if (type == LaunchType.PACKAGE) {
       packages= getListAttribute(configuration, TestNGLaunchConfigurationConstants.PACKAGE_TEST_LIST);
     }
 
+    // ~~
+    // collect test parameters
+    //
+
+    List<IJavaElement> je = Lists.newArrayList();
+    if (packages != null) {
+      for (String pkg : packages) {
+        for (IPackageFragment pf : ijp.getPackageFragments()) {
+          if (pf.getElementName().equals(pkg)) {
+            je.add(pf);
+            break;
+          }
+        }
+      }
+    }
+    if (testClasses != null) {
+      for (String clz : testClasses) {
+        IType t = ijp.findType(clz);
+        if (t != null) {
+          je.add(t);
+        }
+      }
+    }
+    if (classMethods != null) {
+      for (Entry<String, List<String>> entry : classMethods.entrySet()) {
+        IType t = ijp.findType(entry.getKey());
+        for (String mthd : entry.getValue()) {
+          je.add(t.getMethod(mthd, null));
+        }
+      }
+    }
+
+    Map<String, String> parameters= ParameterSolver.solveParameters(je.toArray(new IJavaElement[0]));

```

Comment:
collect the test parameters each time on launching the tests for the following reasons:
1. original implementation cached the test parameters in the `LaunchConfiguration` when first launch, however, if people change the parameters in their code, the cache not got refreshed.
2. original implementation collect the test parameters only when people first run the testng tests via right click context menu, however, if people run the test by creating a new Launch configuration manually, the test parameters not got collected.